packages/babel-parser/src/parser/expression.ts

```diff
@@ -1918,7 +1918,10 @@ export default abstract class ExpressionParser extends LValParser {
     node.callee = this.parseNoCallExpr();
     if (node.callee.type === "Import") {
       this.raise(Errors.ImportCallNotNewExpression, { at: node.callee });
-    } else if (this.isOptionalChain(node.callee)) {
+    } else if (
+      this.isOptionalChain(node.callee) &&
+      !node.callee.extra?.parenthesized
```

Comment:
Yes, this test should throw an exception, but it doesn't.
But unfortunately I didn't find a good way to fix it.
The ast of ts is different from ours. It splits this statement into `call(non-null(prop-access(new,?.xxx)))`.

Reply:
Maybe we can move the error reporting here?

https://github.com/babel/babel/blob/41aac8b8037c566047a1f6792cc81c0e01613373/packages/babel-parser/src/parser/expression.ts#L771

When `noCalls` is `true`, we should throw the error anyway. If the next token is `(` we stop otherwise we continue for the error recovery mode. As a bonus, we don't have to deal with `extra?.parenthesized` (because it will be descended from `parseExprAtom` instead) and it should work with non null TS expression, too.